### ISSUE TITLE
Stack overflow in sign() when domain complex

### ISSUE THREAD

**Issue by rtoy** [rtoy/maxima#607]
Imported from SourceForge on 2024-07-03 02:08:43
Created by **[ricardo-l](https://sourceforge.net/u/ricardo-l/)** on 2017-08-12 21:10:40
Original: https://sourceforge.net/p/maxima/bugs/3324

---

Maxima: 5.39.0 2017-07-24 14:01:48
OS:x86_64-unknown-linux-gnu
ECL:16.1.2

Copying from discussion at this thread https://groups.google.com/forum/#!topic/sage-devel/teLBaNipBDM, 

The simplest example I (Robert Dodier) found is this:
~~~~
(%i2) domain:complex;
(%o2) complex
(%i3) declare([a,b,c], real);
(%o3) done
(%i4) assume(a>0, b>0, c>0);
(%o4) [a > 0,b > 0,c > 0]
(%i5) sign(c - b^(1/3)*c^(1/3));
(%o5) pnz
(%i6) sign(c - (b*c)^(1/3));
[stack overflow here]
~~~~
Note that `(b*c)^(1/3) --> b^(1/3)*c^(1/3)` when domain = real but not
domain = complex.

Apparently, sign() and simplification interact to produce an infinite sequence of calls.

**Comment by rtoy** [comment]
Imported from SourceForge on 2024-07-03 02:08:45
Created by **[robert_dodier](https://sourceforge.net/u/robert_dodier/)** on 2022-02-27 23:55:48
Original: https://sourceforge.net/p/maxima/bugs/3324/#6573

---

- **status**: open --> closed

**Comment by rtoy** [comment]
Imported from SourceForge on 2024-07-03 02:08:49
Created by **[robert_dodier](https://sourceforge.net/u/robert_dodier/)** on 2022-02-27 23:55:48
Original: https://sourceforge.net/p/maxima/bugs/3324/#e8dc

---

This bug appears to have been fixed between Maxima 5.43 and 5.44, although after a brief look at the commit log, I don't know what commit it was. I put the example shown above in the test collection (commit 341b46) to hopefully ensure the bug stays fixed. Closing this report as fixed.
